src/k8s/cmd/k8s/k8s_join_cluster.go

```diff
@@ -47,14 +51,29 @@ var (
 				return fmt.Errorf("failed to create cluster client: %w", err)
 			}
 
-			err = client.JoinNode(cmd.Context(), joinNodeCmdOpts.name, joinNodeCmdOpts.address, token)
+			if c.IsBootstrapped(cmd.Context()) {
+				if joinNodeCmdOpts.force {
+					setup.CleanupNode(cmd.Context(), c.Mc, joinNodeCmdOpts.name)
+				} else {
+					errMsg := "A k8s cluster is already running on this node.\n" +
+						"If you want to remove the current cluster and join the other one, rerun this command with `--force`\n" +
+						"Warning: This will unrecoverably remove all data on this node!"
+					return fmt.Errorf(errMsg)
+				}
+			}
+
+			err = c.JoinNode(cmd.Context(), joinNodeCmdOpts.name, joinNodeCmdOpts.address, token)
 			if err != nil {
 				return fmt.Errorf("failed to join cluster: %w", err)
 			}
 
-			// TODO: (with flag) wait until the node has joined the cluster (e.g. node appears on Kubernetes)
-
 			fmt.Println("Joined the cluster.")
+
+			if joinNodeCmdOpts.waitReady {
+				fmt.Println("Waiting for the node to become ready.")
+				utils.WaitForNodeToBeReady(cmd.Context(), c.Mc, joinNodeCmdOpts.name)
```

Comment:
this needs exposing the microcluster client on the package, so that we can call the `Mc.GetClusterMembers()` directly. We have already added our `client.ClusterStatus()`, should the role just be included there instead?

Reply:
ah, it is already? https://github.com/canonical/k8s-snap/blob/main/src/k8s/api/v1/types.go#L54